src/pystow/utils.py

```diff
@@ -306,35 +305,31 @@ def download_from_google(
     :param clean_on_failure: If true, will delete the file on any exception raised during download
     :param hexdigests:
         The expected hexdigests as (algorithm_name, expected_hex_digest) pairs.
+
+    :raises Exception: Thrown if an error besides a keyboard interrupt is thrown during download
+    :raises KeyboardInterrupt: If a keyboard interrupt is thrown during download
     """
-    # TODO: Reduce code duplication with download
-    # input normalization
     path = Path(path).resolve()
 
-    if os.path.exists(path) and not force:
+    if path.exists() and not force:
         raise_on_digest_mismatch(path=path, hexdigests=hexdigests)
-        logger.debug('did not re-download %s for ID %s', path, file_id)
+        logger.debug('did not re-download %s from %s', path, file_id)
```

Comment:
`file_id` is not a URL.

Reply:
oh, I accidentally clobbered your updated text here too. soz